### PR TITLE
Fixes advanced proc calls not working + various code cleaning

### PR DESCRIPTION
**Pull request by Menshin**
## About The Pull Request

Fixes advances proccall not working as in #42937, but also handles the `target_selected` (a.k.a non global) case .
Various code cleaning while i was at it.

Closes #42937.

:cl: monster860
fix: fixes advanced proccall
/:cl:

### CODE REVIEW

File: code/modules/admin/verbs/debug.dm [hunk 1]
```diff
@@ -108,8 +104,8 @@ GLOBAL_LIST_EMPTY(AdminProcCallSpamPrevention)
 GLOBAL_PROTECT(AdminProcCallSpamPrevention)
 
 /proc/WrapAdminProcCall(datum/target, procname, list/arguments)
-	if(target && procname == "Del")
```
Comment (AnturK) [inline code/modules/admin/verbs/debug.dm]: Why this change ? Destroy is completely unrelated.

**Review by AnturK — DISMISSED** [review]

File: code/modules/admin/verbs/debug.dm (hunk 1 above)
Comment (Menshin) [reply code/modules/admin/verbs/debug.dm]: Because `Destroy()` shouldn't be called by anything but `qdel()`.
It's reverted though.

Comment (AnturK) [reply code/modules/admin/verbs/debug.dm]: In code sure,  but this is debugging tool. Del is only here to prevent privilege escalation.

Comment (Menshin) [reply code/modules/admin/verbs/debug.dm]: Fair enough, changes done btw.
